### PR TITLE
[release/1.3.1] [DEVOPS-398] Testnet config

### PR DESCRIPTION
**Pull request by rvl**
## Description

This PR contains:

* Part of testnet configuration, cherry-picked from #3091 and #3243 (develop branch).
* Docs with an explanation of testnet parameters.
* A fix for CBR-414, cherry-picked from #3582, so that "testnet" and "mainnet" demo clusters can both be launched for testing.

## Linked issues

https://iohk.myjetbrains.com/youtrack/issue/DEVOPS-398
https://iohk.myjetbrains.com/youtrack/issue/CO-354
https://iohk.myjetbrains.com/youtrack/issue/CBR-414

## QA Steps

    nix-build -A demoClusterLaunchGenesis -o demo-cluster-launch-genesis.sh
    ./demo-cluster-launch-genesis.sh

### CODE REVIEW

File: lib/configuration.yaml [hunk 1]
```diff
@@ -14869,21 +14873,28 @@ testnet_launch: &testnet_launch
         <<: *mainnet_base_spec
         protocolConstants:
           <<: *mainnet_base_protocolConstants
-          protocolMagic: 1097911063
-        avvmDistr: *mainnet_avvmDistr
+          protocolMagic:
+            pm: 1097911063
+            requiresNetworkMagic: NMMustBeJust
```
[the remark this answers is not in the document]
Comment (intricate) [reply lib/configuration.yaml]: Also wondering about this..

[the remark this answers is not in the document]
Comment (rvl) [reply lib/configuration.yaml]: Thanks for :question:. This section describes the testnet genesis parameters. The `<< mainnet_...` means to "inherit" config from the section labelled `&mainnet_...`. So my intention is to set up the testnet genesis data with most of the same parameters as mainnet, except overriding things such as `protocolMagic`.

File: lib/configuration.yaml [hunk 2]
```diff
@@ -51,7 +51,9 @@ dev: &dev
           unlockStakeEpoch: 18446744073709551615 # last epoch (maxBound @Word64)
         protocolConstants: &dev_core_genesis_spec_protocolConstants
           k: 2
-          protocolMagic: 55550001
+          protocolMagic:
+            pm: 55550001
+            requiresNetworkMagic: NMMustBeNothing
```
[the remark this answers is not in the document]
Comment (rvl) [reply lib/configuration.yaml]: Yes I would like any dev clusters to work more like mainnet than testnet.

File: lib/configuration.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhuesch) [reply lib/configuration.yaml]: Ah ok, that makes sense. Thanks for clarifying.

**Comment by rvl** [comment]
Thanks @dcoutts . I have reset the `mainnet_base` genesis spec `protocolMagic`, so that there is no double-overriding of the setting. In addition, I made all of the `testnet_launch` genesis spec `protocolConstants` explicit and not inherited just to try and improve clarity.

**Comment by jmitchell** [comment]
We need to ensure any legacy clusters, like staging and mainnet, explicitly pass `requiresNetworkMagic: NMMustBeNothing`. Otherwise, they will default to `NMMustBeJust`, which is the new address behavior for testnet.

**Pull request merged**